### PR TITLE
Move array_dims in the future manipulator

### PR DESCRIPTION
**Pull request by alkino**
Fix vector<Eigen::Matrix> and boost::multi_array<Eigen::Matrix>, from now
Eigen::Matrix has 2 dimensions and no more 1 with hackish things.

### CODE REVIEW

File: include/highfive/bits/H5Utils.hpp [hunk 1]
```diff
@@ -38,63 +38,90 @@ class FixedLenStringArray;
...
-
-// determine at compile time number of dimensions of in memory datasets
 template <typename T>
-struct array_dims {
-    static constexpr size_t value = 0;
+struct manipulator {
+    using type = T;
+
+    static const size_t n_dims = 0;
+    static const size_t r_n_dims = n_dims;
```
Comment (tdegeus) [inline include/highfive/bits/H5Utils.hpp:46]: To me, the variable names are not really descriptive...

**Review by tdegeus — COMMENTED** [review]
Very nice that you are moving all details to a central location. This will simplify the lib in the future.

File: include/highfive/bits/H5Utils.hpp (hunk 1 above)
Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:46]: Help me with that, please, I'm not good.

Comment (tdegeus) [reply include/highfive/bits/H5Utils.hpp:46]: Some suggestions:

* `n_dims` -> `rank` / `dimension` or `outer_rank` / `outer_dimension` 
* `r_n_dims` -> `recursive_rank` / `total_rank` / `full_rank` / `*_dimension` or `rank` / `dimension`

Or... something even better ;) I'd be happy to iterate.

Comment (ferdonline) [reply include/highfive/bits/H5Utils.hpp:46]: Maybe also
 `manipulator` -> `array_info` ?
Or you intent that in the future to manipulate something? (but the word is not much used I guess)

**Comment by ferdonline** [comment]
I like this improvement! Thanks @alkino

File: include/highfive/bits/H5Utils.hpp (hunk 1 above)
Comment (ferdonline) [reply include/highfive/bits/H5Utils.hpp:46]: for the properties, I would keep consistency. In DataSpace we have the following methods:
```
size_t getNumberDimensions() const;
size_t getElementCount() const;
```
so, what about
`.number_dimensions`
`.element_count`

Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:46]: `n_dims` => number of dimensions of the Object itself (e.g `std::vector` => 1)
`r_n_dims` => total number of dimensions (e.g `std::vector<std::vector>` => 2),
It's not an  `element_count`

Comment (alkino) [reply include/highfive/bits/H5Utils.hpp:46]: it's not only `array_info`, i  will add methods to resize array, `ŧransform_read`, `transform_write`, etc...

**Review by alexsavulescu — COMMENTED** [review]
LGTM, nice work.

**Review by ferdonline — APPROVED** [review]

**Pull request merged**